dmutils/content_loader.py

```diff
@@ -419,17 +398,52 @@ def _get_single_question_data(self, form_data):
 
         return {self.id: value}
 
+    def get_error_message(self, message_key, field_name=None):
+        """Return a single error message.
+
+        :param message_key: error message key as returned by the data API
+        :param field_name:
+        """
+        if field_name is None:
+            field_name = self.id
+        for validation in self.get_question(field_name).get('validations', []):
+            if validation['name'] == message_key:
+                if validation.get('field', field_name) == field_name:
+                    return validation['message']
+        return 'There was a problem with the answer to this question'
+
     @property
     def label(self):
         return self.get('name') or self.question
 
     @property
     def form_fields(self):
-        if self.type == 'pricing':
-            return PRICE_FIELDS
+        if self.get('fields', False):
+            return self.get('fields').values()
         else:
+            # pricing fields should have fields.
+            # throw an assertion error if they don't.
+            # TODO: maybe we can check this elsewhere?
+            assert self.type != "pricing"
```

Comment:
Agreed, this assertion is intended to help debugging rather than runtime safety, until we find a better way of handling it explicitly.